binutils: Replace boolean switch with if statement

For true boolean values, you should just use if-else statement, 
not switch. Else, you'll get this error:

error: switch condition has boolean value [-Werror=switch-bool]

Signed-off-by: Daniel Dragomir <daniel.dragomir@windriver.com>

Index: binutils-2.23.2/bfd/elf64-alpha.c
===================================================================
--- binutils-2.23.2.orig/bfd/elf64-alpha.c
+++ binutils-2.23.2/bfd/elf64-alpha.c
@@ -3585,11 +3585,9 @@ elf64_alpha_relax_tls_get_addr (struct a
   use_gottprel = FALSE;
   new_symndx = is_gd ? ELF64_R_SYM (irel->r_info) : STN_UNDEF;
 
-  switch (!dynamic && !info->link_info->shared)
+  if (1 == !dynamic && !info->link_info->shared)
     {
-    case 1:
-      {
-	bfd_vma tp_base;
+        bfd_vma tp_base;
 	bfd_signed_vma disp;
 
 	BFD_ASSERT (elf_hash_table (info->link_info)->tls_sec != NULL);
@@ -3605,7 +3603,6 @@ elf64_alpha_relax_tls_get_addr (struct a
 	    irel[0].r_offset = pos[0] - info->contents;
 	    irel[0].r_info = ELF64_R_INFO (new_symndx, R_ALPHA_TPREL16);
 	    irel[1].r_info = ELF64_R_INFO (0, R_ALPHA_NONE);
-	    break;
 	  }
 	else if (disp >= -(bfd_signed_vma) 0x80000000
 		 && disp < (bfd_signed_vma) 0x7fff8000
@@ -3620,12 +3617,12 @@ elf64_alpha_relax_tls_get_addr (struct a
 	    irel[0].r_info = ELF64_R_INFO (new_symndx, R_ALPHA_TPRELHI);
 	    irel[1].r_offset = pos[1] - info->contents;
 	    irel[1].r_info = ELF64_R_INFO (new_symndx, R_ALPHA_TPRELLO);
-	    break;
 	  }
       }
       /* FALLTHRU */
 
-    default:
+  else
+    {
       use_gottprel = TRUE;
 
       insn = (OP_LDQ << 26) | (tlsgd_reg << 21) | (29 << 16);
@@ -3635,7 +3632,6 @@ elf64_alpha_relax_tls_get_addr (struct a
       irel[0].r_offset = pos[0] - info->contents;
       irel[0].r_info = ELF64_R_INFO (new_symndx, R_ALPHA_GOTTPREL);
       irel[1].r_info = ELF64_R_INFO (0, R_ALPHA_NONE);
-      break;
     }
 
   bfd_put_32 (info->abfd, (bfd_vma) INSN_RDUNIQ, pos[2]);
